### ISSUE TITLE
collaborativefolders_delete_instance could be broken

### ISSUE THREAD

**Issue by Dagefoerde** [learnweb/moodle-mod_collaborativefolders#16]
I once created and deleted a single instance. Afterwards, the admin's activities overview page showed mod_collaborativefolders with 1 instance, although there should be none. It is possible that `collaborativefolders_delete_instance` is broken. 

I am actually unsure whether https://github.com/learnweb/moodle-mod_collaborativefolders/blob/fdd74c053f4dc10a0a558f8fe3a830eac0496f7b/lib.php#L138 works as intended
```php
    if (!$collaborativefolders = $DB->get_record('collaborativefolders', array('id' => $id))) {
```
not sure whether `!` negates the full thing or only the assigned variable!

[original reporter: @Dagefoerde]

**Comment by davosmith** [comment]
It is better just to call delete_records, then return true.

If delete_records has an error, it will throw an exception, otherwise it always returns true.

If the record doesn't exist, we have nothing to delete, so we can safely return true, as the clean-up has already been done.

I've a patch locally to fix this, so I'll open a pull request when testing is done.

**Comment by Dagefoerde** [comment]
Fixed via #34
